x-pack/plugins/fleet/public/applications/fleet/app.tsx

```diff
@@ -141,21 +140,6 @@ export const WithPermissionsAndSetup: React.FC = memo(({ children }) => {
         const permissionsResponse = await sendGetPermissionsCheck();
         setIsPermissionsLoading(false);
         if (permissionsResponse.data?.success) {
-          try {
-            const setupResponse = await sendSetup();
-            if (setupResponse.error) {
-              setInitializationError(setupResponse.error);
-            }
-            if (setupResponse.data?.nonFatalErrors?.length) {
-              notifications.toasts.addError(setupResponse.data.nonFatalErrors[0], {
-                title: i18n.translate('xpack.fleet.setup.uiPreconfigurationErrorTitle', {
-                  defaultMessage: 'Configuration error',
-                }),
-              });
-            }
-          } catch (err) {
-            setInitializationError(err);
-          }
```

Comment:
One thing I've been thinking is that maybe we should keep this around for a really hacky built-in retry in case anything fails. It would also serve the purpose of surfacing the error message. Definitely other ways to accomplish this though.

Reply:
Yeah I think this is fair. We talked about optimizing the "no-op" flow fr the `/setup` endpoint to ensure we're not incurring too much network overhead by calling this on the Fleet application's page load lifecycle. I think I'll restore this for now and include changes to the `/setup` API as above.